### PR TITLE
Update to latest common library

### LINKED ISSUE #23: Update to latest standards
This issue will cover multiple PRs
- [x] new CI / latest php
- [x] keboola coding standards
- [x] db-extractor-common 9.*
- [x] json configurations
The above is covered by https://github.com/keboola/db-extractor-pgsql/pull/29
- [x] row configurations
- [ ] datadir testing

### PR DESCRIPTION
**Pull request by pivnicek**
Fixes: #2 #19 and #22 
Partially Fixes #23

### CODE REVIEW

File: src/Keboola/DbExtractor/Extractor/PgSQL.php [hunk 1]
```diff
@@ -93,10 +100,10 @@ public function export(array $table)
                         throw $applicationException;
                     }
                 } else {
-                    $csvCreated = $this->executeQueryPDO($query, $this->createOutputCsv($outputTable));
+                    $csvCreated = $this->executeQueryPDO($query, $this->createOutputCsv($outputTable), $advancedQuery);
                 }
                 break;
-            } catch (\PDOException $e) {
+            } catch (PDOException $e) {
```
Comment (Actimel) [inline src/Keboola/DbExtractor/Extractor/PgSQL.php]: Use `\PDOException`.

File: src/Keboola/DbExtractor/Extractor/PgSQL.php [hunk 2]
```diff
@@ -114,16 +121,18 @@ public function export(array $table)
...
         }
 
-        return $outputTable;
+        return [
+            "outputTable"=> $outputTable,
+        ];
     }
 
-    protected function executeQueryPDO($query, CsvFile $csv)
+    protected function executeQueryPDO(string $query, CsvFile $csv, bool $advancedQuery): bool
```
Comment (Actimel) [inline src/Keboola/DbExtractor/Extractor/PgSQL.php]: Return type should be `void`.

File: src/Keboola/DbExtractor/Extractor/PgSQL.php [hunk 3]
```diff
@@ -155,21 +167,26 @@ protected function executeQueryPDO($query, CsvFile $csv)
...
+            } catch (Throwable $e2) {
             }
             $innerStatement = null;
             $stmt = null;
             throw $e;
         }
     }
 
-    protected function executeQuery($query, CsvFile $csvFile, $tableName)
+    protected function executeCopyQuery(string $query, CsvFile $csvFile, string $tableName, bool $advancedQuery): bool
```
Comment (Actimel) [inline src/Keboola/DbExtractor/Extractor/PgSQL.php]: Return type should be `void`.

File: src/Keboola/DbExtractor/Extractor/PgSQL.php [hunk 4]
```diff
@@ -248,21 +265,23 @@ function ($table) {
...
+        $arr = $res->fetchAll(PDO::FETCH_ASSOC);
         $tableNameArray = [];
         $tableDefs = [];
         foreach ($arr as $table) {
             $tableNameArray[] = $table['table_name'];
             $tableDefs[$table['table_schema'] . '.' . $table['table_name']] = [
                 'name' => $table['table_name'],
                 'schema' => (isset($table['table_schema'])) ? $table['table_schema'] : null,
-                'type' => (isset($table['table_type'])) ? $table['table_type'] : null
+                'type' => (isset($table['table_type'])) ? $table['table_type'] : null,
```
Comment (Actimel) [inline src/Keboola/DbExtractor/Extractor/PgSQL.php]: How about using `'type' => $table['table_type'] ?? null`.

File: src/Keboola/DbExtractor/Extractor/PgSQL.php [hunk 5]
```diff
@@ -325,7 +345,7 @@ function ($table) {
         return array_values($tableDefs);
     }
 
-    public function simpleQuery(array $table, array $columns = array())
+    public function simpleQuery(array $table, array $columns = array()): string
```
Comment (Actimel) [inline src/Keboola/DbExtractor/Extractor/PgSQL.php]: Use `[]` instead old `array()`.

File: tests/Keboola/Extractor/ApplicationTest.php [hunk 6]
```diff
@@ -9,20 +11,19 @@
...
+    /** @var  string */
     protected $rootPath;
 
-    public function setUp()
+    /** @var string  */
+    protected $dataDir = __DIR__ . '/../../data';
+
+    public function setUp(): void
     {
+        $this->rootPath = '/code/';
```
Comment (MiroCillik) [inline tests/Keboola/Extractor/ApplicationTest.php]: Better to use relative path I think

File: composer.json [hunk 7]
```diff
@@ -1,19 +1,22 @@
...
+        }
+    ],
     "require": {
-        "keboola/db-extractor-common": "~5.2"
+        "keboola/db-extractor-common": "~9"
     },
     "require-dev": {
         "phpunit/phpunit": "4.6.*",
-        "squizlabs/php_codesniffer": "^2.6",
+        "keboola/coding-standard": "^2.0",
```
[the remark this answers is not in the document]
Comment (pivnicek) [reply composer.json]: wow, damn, that moved fast:)

File: src/Keboola/DbExtractor/Extractor/PgSQL.php [hunk 8]
```diff
@@ -92,7 +92,7 @@ public function export(array $table): array
             $csvCreated = true;
         } catch (Throwable $e) {
             // There was an error, so let's try the old method
-            if (get_class($e) !== "ApplicationException") {
+            if (get_class($e) !== "Keboola\DbExtractor\Exception\ApplicationException") {
```
Comment (Actimel) [inline src/Keboola/DbExtractor/Extractor/PgSQL.php]: Better use `if (!$e instanceof ApplicationException)`

File: src/Keboola/DbExtractor/Extractor/PgSQL.php [hunk 9]
```diff
@@ -42,88 +46,105 @@ public function createConnection($dbParams)
...
+            }
             try {
-                if ($tries > 0) {
-                    $this->logger->info("Retrying query");
-                    $this->restartConnection();
-                }
-                if (!$copyFailed) {
+                // recreate the db connection
+                $this->restartConnection();
+            } catch (Throwable $e) {
```
Comment (odinuv) [inline src/Keboola/DbExtractor/Extractor/PgSQL.php]: different $e

File: src/Keboola/DbExtractor/Extractor/PgSQL.php [hunk 10]
```diff
@@ -42,88 +46,105 @@ public function createConnection($dbParams)
...
+                        throw $e;
                     }
-                } else {
-                    $csvCreated = $this->executeQueryPDO($query, $this->createOutputCsv($outputTable));
-                }
-                break;
-            } catch (\PDOException $e) {
-                $exception = new UserException("DB query [{$table['name']}] failed: " . $e->getMessage(), 0, $e);
+                });
+            } catch (PDOException $e) {
```
Comment (odinuv) [inline src/Keboola/DbExtractor/Extractor/PgSQL.php]: dtto and the next one too

File: src/Keboola/DbExtractor/Extractor/PgSQL.php [hunk 11]
```diff
@@ -42,88 +46,105 @@ public function createConnection($dbParams)
...
+        try {
+            $this->executeCopyQuery(
+                $query,
+                $this->createOutputCsv($outputTable),
+                $table['name'],
+                $advancedQuery
+            );
+            $csvCreated = true;
+        } catch (Throwable $e) {
+            // There was an error, so let's try the old method
```
Comment (odinuv) [inline src/Keboola/DbExtractor/Extractor/PgSQL.php]: any way to filter out this https://github.com/keboola/db-extractor-pgsql/issues/28 or leave that for next round?

Comment (pivnicek) [reply src/Keboola/DbExtractor/Extractor/PgSQL.php]: I'm going to first add the SQL codes extension to the RetryProxy, and then we'll solve #28 
Sound reasonable?

File: src/Keboola/DbExtractor/Extractor/PgSQL.php (hunk 10 above)
Comment (pivnicek) [reply src/Keboola/DbExtractor/Extractor/PgSQL.php]: good call on these, I've given them all distinct (and meaningful:) names now

**Comment by pivnicek** [comment]
@odinuv @Actimel jeste neco? nebo muzu ho releasnout?

**Comment by odinuv** [comment]
za me ok, ale spusteny jsem to nemel, mam to jeste nejak zkouset?

**Comment by pivnicek** [comment]
ja myslim ze ne.  zadny novy featury nejsou.

**Pull request merged**